### ISSUE TITLE
ANTLR generates a wrong lexer code if type, channel or mode commands use inproper identifier

### ISSUE THREAD

**Issue by KvanTTT** [antlr/antlr4#1411]
Consider the following simple grammar:

```ANTLR
lexer grammar WrongIdInCommandsLexer;

tokens { TOKEN1 }
channels { CHANNEL1 }

TOKEN: 'asdf'  -> type(CHANNEL1), channel(MODE1), mode(TOKEN1);

mode MODE1;

MODE1_TOKEN: 'bbbb';
```

### Actual behavior

ANTLR generates inproper lexer which throws an NRE exception during parsing process (C# Sharwell runtime).

### Expected behavior

ANTLR should show 3 errors about wrong type, channel and mode identifiers in respective commands and should not generate inproper lexer.

**Comment by sharwell** [comment]
I can't think of a case where it makes sense to reference a constant of a different type than the lexer command supports. So I'm 👍 on making these cases errors.

**Comment by KvanTTT** [comment]
I encountered with this problem during merging two lexer grammars. So, this issue is actual.

**Comment by sharwell** [comment]
I would propose the following specific new errors.

:memo: In #1417 I requested no changes to existing errors to simplify that review. I think it makes sense to add these new error types in a follow-up commit (or even a separate pull request), but wait for @parrt to give feedback on them. I already checked that these error numbers are not in use here or in my fork.

```
error(175): CHANNEL1 is not a recognized token name
error(176): TOKEN1 is not a recognized mode name
error(177): MODE1 is not a recognized channel name
```
